// @chakra-ui/vue/dist/cjs/CCode.js

/*
 * ! @chakra-ui/vue v0.11.0 by Jonathan Bakebwa @codebender828
 * MIT License
 * https://github.com/chakra-ui/chakra-ui-vue
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var utils = require('./index-068c5bef.js');
require('./chakra-ui-styled-system.esm-1a60142c.js');
require('@emotion/css');
require('vue');
require('./color-mode-observer-36841f0d.js');
var props_types = require('./props.types-1910f70e.js');
var badge_styles = require('./badge.styles-478dc3e6.js');

/**
 * Hey! Welcome to @chakra-ui/vue Code
 *
 * Code is a component used to display inline code.
 *
 * @see Docs     https://vue.chakra-ui.com/code
 * @see Source   https://github.com/chakra-ui/chakra-ui-vue/blob/master/packages/chakra-ui-core/src/CCode/CCode.js
 */
/**
 * CCode component
 *
 * Inline code component
 *
 * @extends CBox
 * @see Docs https://vue.chakra-ui.com/code
 */

var CCode = {
  name: 'CCode',
  mixins: [utils.createStyledAttrsMixin('CCode')],
  props: {
    variantColor: {
      type: String,
      default: 'gray'
    },
    fontFamily: {
      type: props_types.SNA,
      default: 'mono'
    }
  },
  computed: {
    badgeStyle: function badgeStyle() {
      utils.useVariantColorWarning(this.theme, 'CCode', this.variantColor);
      return badge_styles.useBadgeStyle({
        variant: 'subtle',
        color: this.variantColor,
        colorMode: this.colorMode,
        theme: this.theme
      });
    },
    componentStyles: function componentStyles() {
      return Object.assign({}, {
        display: 'inline-block',
        fontSize: 'sm',
        px: '0.2em',
        fontFamily: 'mono',
        rounded: 'sm'
      }, this.badgeStyle);
    }
  },
  render: function render(h) {
    return h('code', {
      class: this.className,
      attrs: this.computedAttrs
    }, this.$slots.default);
  }
};

exports.default = CCode;
