// @chakra-ui/vue/dist/cjs/CPseudoBox.js

/*
 * ! @chakra-ui/vue v0.11.0 by Jonathan Bakebwa @codebender828
 * MIT License
 * https://github.com/chakra-ui/chakra-ui-vue
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var utils = require('./index-068c5bef.js');
require('./chakra-ui-styled-system.esm-1a60142c.js');
require('@emotion/css');
require('vue');
require('./color-mode-observer-36841f0d.js');

/**
 * Hey! Welcome to @chakra-ui/vue PseudoBox
 *
 * The PseudoBox component
 *
 * @see Docs     https://vue.chakra-ui.com/pseudobox
 * @see Source   https://github.com/chakra-ui/chakra-ui-vue/blob/master/packages/chakra-ui-core/src/CPseudoBox/CPseudoBox.js
 */
/**
 * CPseudoBox component
 *
 * The pseudobox component that accepts pseudo props
 *
 * @extends CBox
 * @see Docs https://vue.chakra-ui.com/pseudobox
 */

var CPseudoBox = {
  name: 'CPseudoBox',
  mixins: [utils.createStyledAttrsMixin('CPseudoBox')],
  props: {
    as: {
      type: [String, Object],
      default: function _default() {
        return 'div';
      }
    },
    to: [String, Object]
  },
  render: function render(h) {
    return h(this.as, {
      props: {
        to: this.to
      },
      class: this.className,
      on: this.listeners$,
      attrs: this.computedAttrs
    }, this.$slots.default);
  }
};

exports.default = CPseudoBox;
